### ISSUE TITLE
Slow rendering (Poland) on S22

### ISSUE THREAD

**Issue by JakubIgnacyGorski** [osmandapp/OsmAnd#20795]
### Description

OpenGL display runs slowly much slower than version 1. I tested on two phones with exactly the same settings and on the Xiaomi Mi 9 the problem does not occur while on the Samsung Galaxy S22 the map display runs very slowly. I tested the solution to the problem [#19951](https://github.com/osmandapp/OsmAnd/issues/19951) did not work. In the attachment I have added the behavior for Samsung Galaxy S22.

https://github.com/user-attachments/assets/2be6b1cc-cbae-4bbf-9b81-4857fa60e4b7

### Steps to reproduce

Open the maps and try zooming in or panning to a different location.

### Actual result

Very slow loading of the map.

### Expected result

Running faster than version 1 rendering engine.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version:  4.8.6
Android/iOS version: 14 
Device model: Samsung Galaxy S22
```

**Comment by smlpt** [comment]
Not sure if this is related to #19927, #19511 or #18867 and will be closed as duplicate, but I also noticed a pretty sluggish map panning/scrolling with the current version of Osmand.
While my Fairphone 3 is starting to show its age, I still remember Osmand being way snappier a year ago on the same device, so I did the following:
- uninstalled Osmand
- installed an older version (4.4.06) using the Aurora store
- tested the panning performance in the world map without any tweaks to the settings. No other downloads, no live maps, pure default settings.
- I repeated these steps with other versions. With 4.4.X everything is fine, all the way up to 4.7.17. The first version to show sluggish behavior is **4.8.4**, which is the first 4.8.X version available as APK in Aurora / APK Mirror.

Now it would be interesting to test the other 4.8.X versions that came before 4.8.4 for performance to narrow down the changes that happened after 4.7.17 and might cause this issue.

The map tiles appear to load with the same speed (you can sometimes see them popping up over white background), but with the older versions panning performance wasn't greatly influenced by loading the map tiles.

```
Osmand Version: 4.4.6-4.8.6 (all tested)
Androis/iOS version: Android 13, /e/OS 
Device model: Fairphone 3
```

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.9.0#3173m, released: 2024-09-12
Later, I also experienced problems with rendering of some regions on my Samsung (galaxy note 20 ultra). But in nightly, map rendering became much faster. 



<table style="width:100%">
  <tr>
  </tr>
  <tr>
    <td><video src="https://github.com/user-attachments/assets/5374e022-f8d5-4195-b1c7-7689772cd13e"  width="300" height="auto"/> </td>
    <td><video src="https://github.com/user-attachments/assets/e9852120-cac6-4cee-8cf3-152827507a77"  width="300" height="auto"/> </td>
  </tr>
</table>


@JakubIgnacyGorski @smlpt could you please try nightly and see if the rendering remains slow. 
Nightly is available by the link https://osmand.net/docs/versions/nightly_versions#nightly-builds
https://download.osmand.net/latest-night-build/OsmAnd-default.apk

**Comment by smlpt** [comment]
Here are screen captures of version 4.7.17, 4.8.0.1 and 4.9 nightly. The latter two have about the same frame rates (if I'd have to guess, I'd say 5-7?), and the older version is more responsive and I'd say has somewhere between 20-30fps.

https://github.com/user-attachments/assets/98478fec-5c58-4028-8c3a-5517b4e2de63

https://github.com/user-attachments/assets/e3bcb732-25f3-441f-a28c-ce5f2a74b4bb

https://github.com/user-attachments/assets/efbc4c10-4e7f-440f-9f0c-2cd7fc03d5c3

**Comment by DmitryAlexei** [comment]
@smlpt you can enable fps widget (how to: https://osmand.net/docs/user/plugins/development#fps-widget ) and see if your assumptions are correct
For me, FPS is always around 80-120
It is also recommended to use + and - buttons for correct testing, as the map may not update until you release your fingers from the screen.

**Comment by JakubIgnacyGorski** [comment]
> OsmAnd~ 4.9.0#3173m, released: 2024-09-12 Later, I also experienced problems with rendering of some regions on my Samsung (galaxy note 20 ultra). But in nightly, map rendering became much faster.
> 
>  2024-09-12.15.13.12.mp4 
>  2024-09-12.15.17.59.mp4 
> @JakubIgnacyGorski @smlpt could you please try nightly and see if the rendering remains slow. Nightly is available by the link https://osmand.net/docs/versions/nightly_versions#nightly-builds https://download.osmand.net/latest-night-build/OsmAnd-default.apk

Sorry for the late reply, I downloaded nightly version 4.9.0#3213m. I don't see any improvement over the stable version.

**Comment by vshcherb** [comment]
Did you by any chance enable Battery saving mode in development? It drops FPS to 15-18 though it was not present in 4.8 I think

**Comment by vshcherb** [comment]
1. Original issue posted in the top comment is already fixed in 4.9, so should we open another issue for subthread about low FPS? @JakubIgnacyGorski if not please post the video of latest in the comment
2. Did you by any chance enable Battery saving mode in development? It drops FPS to 15-18 though it was not present in 4.8 .

**Comment by JakubIgnacyGorski** [comment]
> 1. Original issue posted in the top comment is already fixed in 4.9, so should we open another issue for subthread about low FPS? @JakubIgnacyGorski if not please post the video of latest in the comment
> 2. Did you by any chance enable Battery saving mode in development? It drops FPS to 15-18 though it was not present in 4.8 .

I checked it on nightly version 5.0.0#3674m and now it's working much faster and it's amazing.
Thank you all very much
